account_invoice_partner_number/tests/test_account_invoice_partner_number.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo.tests import common
+
+
+class TestAccountInvoiceParterNumber(common.TransactionCase):
+    def setUp(self):
+        super(TestAccountInvoiceParterNumber, self).setUp()
+        self.account_invoice = self.env['account.invoice']
+        self.account_model = self.env['account.account']
+        self.account_invoice_line = self.env['account.invoice.line']
+
+        self.invoice_account = self.account_model.search(
+            [('user_type_id',
+              '=',
+              self.env.ref('account.data_account_type_receivable').id
+              )], limit=1)
+        # Instance: Invoice Line
+        self.invoice_line = self.account_invoice_line.create(
+            {'name': 'Test invoice line',
+             'account_id': self.invoice_account.id,
+             'quantity': 1.000,
+             'price_unit': 200.00})
+
+    def test_account_invoice_partner_number(self):
+        invoice = self.account_invoice.create({
+            'partner_id': self.env.ref('base.res_partner_1').id,
```

Comment:
Define `cls.partner` in `setUpClass()`.  In the test method, first assert that `partner.partner_no` is false, create an invoice and assert that `invoice.partner_no` is false, and then update `partner.partner_no` and assert `invoice.partner_no` is `000123`.